### DISCUSSION TITLE
Namespace Comments

### DISCUSSION THREAD

**Discussion by keyseven123** [nebulastream/nebulastream#379] [Polls]
In a lot of places of our code, we have comments highlighting the end of a namespace, e.g., 
```c++
namespace API
{
class WindowAggregation;
using WindowAggregationPtr = std::shared_ptr<WindowAggregation>;
} /// namespace API
```

This was done as there was no support showcasing the namespace name at the closing curly bracket. As Clion supports this feature, the question comes, if we should get rid of the comments.

**Comment by klauck** [comment]
`} // namespace API` would be the formatting as in duckdb (https://github.com/duckdb/duckdb/blob/main/src/parser/column_definition.cpp) and hyrise (https://github.com/hyrise/hyrise/blob/master/src/lib/scheduler/worker.hpp)

**Reply by klauck** [reply]
I read/understand why we use `///`

**Reply by klauck** [reply]
https://github.com/nebulastream/nebulastream-public/discussions/18

**Reply by keyseven123** [reply]
The question is simply if we want to have these comments in general. IMO, they are just a really easy way of adding bugs/wrong comments into the code. Clion supports adding the name of the namespace automatically. Therefore, I do not see the need to have them as fixed comments.

**Reply by klauck** [reply]
If you read code, for example, on github, I find these comment helpful, otherwise I do not care much. Isn't there a way to format namespace comments automatically?

**Reply by keyseven123** [reply]
~~In theory yes. We would then have two different clang-tidy calls. One for checking a larger list and one for fixing some specified comments.~~

~~What do the others @ls-1801 @adrianmichalke @zeuchste @alepping thing about this?~~

This does not work, as `clang-tidy -fix` will put `//` as comment, which we do not allow in our code.

**Reply by alepping** [reply]
> If you read code, for example, on github, I find these comment helpful, otherwise I do not care much. Isn't there a way to format namespace comments automatically?

That is also the main argument from my point of view. However, I mostly read the code in CLion, including PRs. Also, I essentially never scroll to the bottom down, or scan for namespace-closing brackets to better understand the code.
